Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs

```diff
@@ -3086,43 +3228,90 @@ specialZoneFlow estimatedQuotes state = do
+
+
+    defaultQuote = fromMaybe ChooseVehicleController.config $ 
+      if newState.props.isRepeatRide then do 
+        let nYQuotes = filter (\item ->item.providerType == ONUS) quoteList
+            defaultQuote_ = find (\item -> isJust item.serviceTierName && item.serviceTierName == newState.props.repeatRideServiceTierName) nYQuotes
+        if isJust defaultQuote_ 
+          then defaultQuote_
+        else (quoteList !! newState.data.selectedEstimatesObject.activeIndex)
+      else 
+        if showMultiProvider' then 
+          quoteList !! newState.data.selectedEstimatesObject.activeIndex
+        else 
+          let nYQuotes = filter (\item ->item.providerType == ONUS) quoteList -- NY Quotes
+          in nYQuotes !! newState.data.selectedEstimatesObject.activeIndex 
+
+
+    zoneType = getSpecialTag $ case head estimates of
+                  Just entity -> view _specialLocationTag entity
+                  Nothing -> Nothing
+
+    hasToll = any (\item -> maybe false (\fareBreakupList -> isEstimateFareBreakupHastitle fareBreakupList "TOLL_CHARGES") (item ^. _estimateFareBreakup)) estimates
+
+
+    topProviderEstimates = filter (\element -> element.providerType == ONUS) quoteList -- filter the ny provider estimates
+    topProviderCheck = if state.data.currentCityConfig.iopConfig.enable then true else not $ null topProviderEstimates -- if iop is not enabled then show ny provider else show multi provider
+    hasQuotes = not $ null quoteList
+    
+  
+  if hasQuotes && topProviderCheck then do -- if choosing multiple provider is not enabled then only show ny
+    let 
+      _ = unsafePerformEffect $ logEvent state.data.logField "ny_user_quote"
+      nearByDrivers = getNearByDrivers estimates
+      nearByDriversLength = length nearByDrivers
+      _ = runFn2 updatePushInIdMap "EstimatePolling" true
+      quoteList' = map (\quote -> quote{activeIndex = defaultQuote.index}) quoteList
+
+    void $ pure $ updateLocalStage SettingPrice
     logStatus "drivers_available" nearByDriversLength
-    let _ = runFn2 updatePushInIdMap "EstimatePolling" true
-        quoteList = map (\quote -> quote{activeIndex = estimatesInfo.defaultQuote.index}) estimatesInfo.quoteList
+    
 
-    void $ pure $ setValueToLocalStore HAS_TOLL_CHARGES $ show estimatesInfo.hasToll
-    exit $ SelectEstimate state 
+
+    void $ pure $ setValueToLocalStore HAS_TOLL_CHARGES $ show hasToll
+    exit $ SelectEstimate newState 
       { data
-        { specialZoneQuoteList = quoteList
+        { specialZoneQuoteList = quoteList'
         , currentSearchResultType = ESTIMATES
-        , selectedEstimatesObject = estimatesInfo.defaultQuote
+        , selectedEstimatesObject = defaultQuote
         , nearByDrivers = if nearByDriversLength > 0 then Just nearByDriversLength else Nothing
+        , iopState { 
+          showPrefButton = state.data.currentCityConfig.iopConfig.enable && (not (null topProviderEstimates)) && (not newState.props.isRepeatRide)
```

Comment:
Put `state.data.currentCityConfig.iopConfig.enable && (not (null topProviderEstimates)) && (not newState.props.isRepeatRide)` in let